scala/private/rule_impls.bzl

```diff
@@ -190,7 +190,7 @@ DiagnosticsFile: {diagnostics_output}
         scalac_jvm_flags,
         ctx.toolchains["@io_bazel_rules_scala//scala:toolchain_type"].scalac_jvm_flags,
     )
-    if len(BAZEL_VERSION) == 0 and enable_diagnostics_report:  # TODO: Add case for released version of bazel with diagnostics whenever it is released.
+    if False:  # TODO: Add case for released version of bazel with diagnostics whenever it is released.
```

Comment:
I think it's a good idea to leave this branch here nevertheless as a reminder for this to be changed after https://github.com/bazelbuild/bazel/pull/11766 is finally merged. Wdyt @liucijus ?

Reply:
I think it's better to completely remove if statement and add required changes when the Bazel with this feature will be available. I would prefer having commented out `diagnostics_file` parameter with the explanation (maybe also worth registering a tracking issue and leave its number in the comment).

Also please remove unused load `load("@io_bazel_rules_scala_config//:config.bzl", "BAZEL_VERSION")`, which is currently reported by the linter test.

Reply:
Sounds great. Let me know if there's anything else that should be done

Reply:
LGTM, thank you!